src/installers/GDWeaveInstaller.ts

```diff
@@ -0,0 +1,82 @@
+import { InstallArgs, PackageInstaller } from "./PackageInstaller";
+import path from "path";
+import FsProvider from "../providers/generic/file/FsProvider";
+import { addToStateFile } from "./InstallRuleInstaller";
+import FileUtils from "../utils/FileUtils";
+
+const basePackageFiles = ["manifest.json", "readme.md", "icon.png"];
+
+// Adapted from the BepInEx & Lovely installers
+export class GDWeaveInstaller implements PackageInstaller {
+    async install(args: InstallArgs) {
+        const fs = FsProvider.instance;
+        const fileRelocations = new Map<string, string>();
+
+        const winmmSrc = path.join(args.packagePath, "winmm.dll");
+        const winmmDest = args.profile.joinToProfilePath("winmm.dll");
+        await fs.copyFile(winmmSrc, winmmDest);
+        fileRelocations.set(winmmSrc, "winmm.dll");
+
+        for (const item of (await FsProvider.instance.readdir(args.packagePath))) {
+            if (basePackageFiles.includes(item.toLowerCase())) continue;
+
+            const srcPath = path.join(args.packagePath, item);
+            const destPath = args.profile.joinToProfilePath(item);
+            if ((await FsProvider.instance.stat(srcPath)).isFile()) {
+                await FsProvider.instance.copyFile(srcPath, destPath);
+            } else {
+                await FsProvider.instance.copyFolder(srcPath, destPath);
+            }
+        }
+
+        await addToStateFile(args.mod, fileRelocations, args.profile);
```

Comment:
I don't think is needed as we don't need to track any extra state to be able to uninstall this. Also I'm not 100% how GDWeave should react to being disabled, but on a quick glance seems state isn't used for that either?

Reply:
Yeah, I wasn't sure how to use the state file, but knowing that there's a better way to accomplish it I'll be moving towards it.